playbooks/manage_edxapp_users_and_groups.yml

```
@@ -197,7 +197,7 @@
         {% if item.get('permissions', []) | length %}--permissions {{ item.permissions | default([]) | map('quote') | join(' ') }}{% endif %}
         {% if item.get('remove') %}--remove{% endif %}
       with_items: "{{ django_groups }}"
-      when: (not group_environment) or group_environment in item.environments
+      when: (service | bool) and (not group_environment) or group_environment in item.environments
```

According to multiple sources that I found ([source1](https://stackoverflow.com/questions/36912726/ansible-how-to-test-that-a-registered-variable-is-not-empty), [source2](https://github.com/ansible/ansible-lint/issues/457#issuecomment-455038475)), the expression `var | bool` would evaluate for `False`, even if var is a non-empty string.

The most robust method which is recommended is 
```suggestion
      when: (service | length > 0) and (not group_environment) or group_environment in item.environments
```

---

@kaustavb12 Good catch! Changed.